CaloFilters/src/CaloCosmicCalib_module.cc

```diff
@@ -90,23 +143,105 @@ namespace mu2e
       if( (energy >= _minenergy) && 
 	  (energy <= _maxenergy) && 
 	  (clsize >= _minncrystalhits) ) {
-        retval = true;
-        ++_npass;
-        // Fill the trigger info object
-        if (trig_ind == 0){
-	  triginfo->_triggerBits.merge(TriggerFlag::caloCalib);
-	  triginfo->_triggerPath = _trigPath;
+	// ----------------------------------------------
+	// Loop over the hits of the selected cluster
+	// -----------------------------------------------
+	caloClusterHits = &cl.caloHitsPtrVector();
+
+	int selflag=0;	
+	int NumMipCel= 0;
+	float Eouter = 0.;
+	float Einner = 0.;
+	float DXmax, DYmax;
+	float RMax=-999;
+	float RMin=-999;
+	float XMin= 999.;
+	float XMax=-999.;	
+	float YoutMin= 999.;
+	float YoutMax=-999.;
+	
+	float xval,yval,rval;
+	for( int icry=0; icry< clsize; icry++){
+	  hitcalo = &(*caloClusterHits->at(icry));	  
+	  float ene   = hitcalo->energyDep();
+	  float time  = hitcalo->time();	  
+	  int _cryid = hitcalo->crystalID();	  
+	  int DiskId = _calogeom->crystal(_cryid).diskID();	
+ 
+	  CLHEP::Hep3Vector crystalpos =
+	    _calogeom->geomUtil().mu2eToDiskFF(DiskId,_calogeom->crystal(_cryid).position());
+	  
+	  if( ene> EMIPcut){         // Use cells with energy above a MIP-like threshold
+	    xval = crystalpos.x();
+	    yval = crystalpos.y();
+	    rval = sqrt(pow(xval,2)+ pow(yval,2));
+	    // ----  find the cell with the highest and lowest Y ---------
+	    if( yval >  YoutMax ){ // Ymax
+	      YoutMax = yval;
+	      XMax = xval;
+	      RMax = rval; } 	   
+	    if( yval < YoutMin ){  // Ymin
+	      YoutMin = yval;
+	      XMin = xval;
+	      RMin = rval; }
+	    // calculate Outer/Inner Ring Energies
+	    
+	    if( rval > RMAXcut ) Eouter = Eouter+ene; // Oouter ring 
+	    if( rval < RMINcut ) Einner = Einner+ene; // Inner  ring 
+	    NumMipCel++; // Increment number of MIP cells
+	    
+	  }
+	  
+	  if(_debug>2) std::cout << moduleDescription().moduleLabel() << " Nc " << numcluster << " cel " << icry << " T-E " << time  << " " << ene << " icry " <<   _cryid  <<  " " << DiskId << " X-Y " << xval << " " << yval << std::endl;
+	  
+	  
+	} // end loop over hits/cluster
+	//====================================
+	// Now fill up the selection flag  ...
+	//====================================
+	if( Eouter>EMIPcut && Einner>EMIPcut && NumMipCel>_mincelinout ) selflag=selflag+1;   // Out-IN or IN-Out
+	
+	if( NumMipCel>_minceldiag && YoutMax>-999. && YoutMin<999. ){	  
+	  DXmax = abs(XMax-XMin);
+	  DYmax = abs(YoutMax-YoutMin);
+	  float DiagVal=0.;
+	  if( DXmax> 0 ) DiagVal = DYmax/DXmax;	  
+	  if( DXmax  < 35.0  ) selflag = selflag+100;  // Vertical selection
+	  if( DiagVal > 1.5 && DiagVal < 2.5 )
```

Comment:
What are these numbers?  Could they ever change?  If so, they should be fhicl parameters.

Reply:
these numbers are now fcl parameters

Reply:
I think a check of DXmax doesn't hurt; you could have a cosmic crossing 3 crystals, out of which 2 have the same X and the 3 doesn't make a hit for whatever reason. It has a very small probability for sure, but I think is better to avoid nan filled in the histograms.